src/lib/table/table-data-source.ts

```diff
@@ -176,9 +176,14 @@ export class MatTableDataSource<T> extends DataSource<T> {
    */
   _updateChangeSubscription() {
     // Sorting and/or pagination should be watched if MatSort and/or MatPaginator are provided.
+    // The events should emit whenever the component emits a change or initializes.
     // Otherwise, use an empty observable stream to take their place.
-    const sortChange: Observable<Sort> = this._sort ? this._sort.sortChange : empty();
-    const pageChange: Observable<PageEvent> = this._paginator ? this._paginator.page : empty();
+    const sortChange: Observable<Sort|void> = this._sort ?
```

Comment:
You could do 
```ts
const sortChange: Observable<Sort> = this._sort ?
    merge<Sort>(this._sort.sortChange, this._sort.initialized) :
    empty() as Observable<Sort>;
```
Which is still valid since empty will never emit anything (similar elsewhere)